evennia/contrib/tests.py

```diff
@@ -1214,6 +1238,199 @@ def test_tbrangefunc(self):
+        self.assertTrue(attacker.db.Combat_ActionsLeft == 0)
+        self.assertTrue(attacker.db.Combat_LastAction == "Test")
+        # Initialize for combat
+        attacker.db.Combat_ActionsLeft = 983
+        turnhandler.initialize_for_combat(attacker)
+        self.assertTrue(attacker.db.Combat_ActionsLeft == 0)
+        self.assertTrue(attacker.db.Combat_LastAction == "null")
+        # Start turn
+        defender.db.Combat_ActionsLeft = 0
+        turnhandler.start_turn(defender)
+        self.assertTrue(defender.db.Combat_ActionsLeft == 1)
+        # Next turn
+        turnhandler.db.fighters = [attacker, defender]
+        turnhandler.db.turn = 0
+        turnhandler.next_turn()
+        self.assertTrue(turnhandler.db.turn == 1)
+        # Turn end check
+        turnhandler.db.fighters = [attacker, defender]
+        turnhandler.db.turn = 0
+        attacker.db.Combat_ActionsLeft = 0
+        turnhandler.turn_end_check(attacker)
+        self.assertTrue(turnhandler.db.turn == 1)
+        # Join fight
+        joiner = create_object(tb_items.TBItemsCharacterTest, key="Joiner")
+        turnhandler.db.fighters = [attacker, defender]
+        turnhandler.db.turn = 0
+        turnhandler.join_fight(joiner)
+        self.assertTrue(turnhandler.db.turn == 1)
+        self.assertTrue(turnhandler.db.fighters == [joiner, attacker, defender])
+        # Remove the script at the end
+        turnhandler.stop()
+        # Now time to test item stuff.
+        user = create_object(tb_items.TBItemsCharacterTest, key="User")
+        testroom = create_object(DefaultRoom, key="Test Room")
+        user.location = testroom
+        test_healpotion = create_object(key="healing potion")
+        test_healpotion.db.item_func = "heal"
+        test_healpotion.db.item_uses = 3
+        # Spend item use
+        tb_items.spend_item_use(test_healpotion, user)
+        self.assertTrue(test_healpotion.db.item_uses == 2)
+        # Use item
+        user.db.hp = 2
+        tb_items.use_item(user, test_healpotion, user)
+        self.assertTrue(user.db.hp > 2)
+        # Add contition
+        tb_items.add_condition(user, user, "Test", 5)
+        self.assertTrue(user.db.conditions == {"Test":[5, user]})
+        # Condition tickdown
+        tb_items.condition_tickdown(user, user)
+        self.assertTrue(user.db.conditions == {"Test":[4, user]})
+        # Test item functions now!
+        # Item heal
+        user.db.hp = 2
+        tb_items.itemfunc_heal(test_healpotion, user, user)
+        # Item add condition
+        user.db.conditions = {}
+        tb_items.itemfunc_add_condition(test_healpotion, user, user)
+        self.assertTrue(user.db.conditions == {"Regeneration":[5, user]})
+        # Item cure condition
+        user.db.conditions = {"Poisoned":[5, user]}
+        tb_items.itemfunc_cure_condition(test_healpotion, user, user)
+        self.assertTrue(user.db.conditions == {})
+        # Delete the test character
```

Comment:
This kind of cleanup belongs in a `tearDown` method, it will then be called after automatically after each test-method in the class.